pkg/controller/cluster/cluster_controller.go

```diff
@@ -356,7 +366,91 @@ func (c *Controller) processNextWorkItem() bool {
 	return true
 }
 
+func (c *Controller) buildRemoteClusterClient(cluster *clusteroperator.Cluster) (clusterapiclient.Interface, error) {
+
+	secretName := cluster.Name + "-kubeconfig"
+
+	secret, err := c.kubeClient.CoreV1().Secrets(cluster.Namespace).Get(secretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	secretData := secret.Data["kubeconfig"]
+
+	config, err := clientcmd.Load(secretData)
+	if err != nil {
+		return nil, err
+	}
+
+	kubeConfig := clientcmd.NewDefaultClientConfig(*config, &clientcmd.ConfigOverrides{})
+	restConfig, err := kubeConfig.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	clusterAPIClient, err := clusterapiclient.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+	return clusterAPIClient, nil
+}
+
+func (c *Controller) ensureRemoteMachineSetsAreDeleted(cluster *clusteroperator.Cluster) error {
```

Comment:
I have some concern that trying to run operations on a remote cluster to which the connection is slow or inaccessible will adversely effect the ability of the controller to timely deal with other tasks. Is there a way that we can run these remote operations in a separate goroutine and have the controller pick up the results later? I am fine tackling that as later work.

Reply:
Sgtm, I'll run these in their own goroutine.